libqtile/utils.py

```diff
@@ -243,16 +234,41 @@ def send_notification(title, message, urgent=False, timeout=10000, id=None):
     passed when calling this function again to replace that notification. See:
     https://developer.gnome.org/notification-spec/
     """
-    if _can_notify and Notify.get_server_info()[0]:
-        notifier = Notify.Notification.new(title, message)
-        if urgent:
-            notifier.set_urgency(Notify.Urgency.CRITICAL)
-        notifier.set_timeout(timeout)
-        if id is None:
-            id = randint(10, 1000)
-        notifier.set_property('id', id)
-        notifier.show()
-        return id
+    id = randint(10, 1000) if id is None else id
+    urgency = 2 if urgent else 1
+
+    try:
+        asyncio.create_task(_notify(title, message, urgency, timeout, id))
+    except RuntimeError:
+        logger.warning("Eventloop has not started. Cannot send notification.")
+
+    return id
+
+
+async def _notify(title, message, urgency, timeout, id):
+    bus = await MessageBus(bus_type=BusType.SESSION).connect()
+
+    msg = Message(destination="org.freedesktop.Notifications",
```

Comment:
Just realised it might be worth refactoring this as it shares some code with the `_send_dbus_message` code for subscribing to signals. No time today to do it.